utils/sane-backends/files/saned.init

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh /etc/rc.common
+
+START=90
+STOP=10
+USE_PROCD=1
+
+sanedconf() {
+	local cfg="$1"
+	config_get user "$cfg" user nobody
+	config_get portrange "$cfg" portrange
+	config_get acl "$cfg" acl
+}
+
+start_service() {
+	local user hplip portrange acl
+
+	. /lib/functions.sh
+
+	config_load saned
+	config_foreach sanedconf saned
```

Comment:
On 16-03-25 12:26 AM, Luiz Angelo Daros de Luca wrote:
> In utils/sane-backends/files/saned.init
> <https://github.com/openwrt/packages/pull/2543#discussion_r57418439>:
>
>> +USE_PROCD=1
>> +
>> +sanedconf() {
>> +	local cfg="$1"
>> +	config_get user "$cfg" user nobody
>> +	config_get portrange "$cfg" portrange
>> +	config_get acl "$cfg" acl
>> +}
>> +
>> +start_service() {
>> +	local user hplip portrange acl
>> +
>> +	. /lib/functions.sh
>> +
>> +	config_load saned
>> +	config_foreach sanedconf saned
>
> I'm no procd expert but foreach seems to be for daemons with repeatable
> configurations. It does not seem to be your case. AFAIK, your script
> support only one "config saned".

To not use foreach you have to make the section a named section - for 
anonymous section you have to use foreach even if there is only one.

Regards,

Daniel